packages/nextjs-mf/src/plugins/NextFederationPlugin/regex-equal.ts

```diff
@@ -1,16 +1,16 @@
 import { RuleSetConditionAbsolute } from 'webpack';
 
 /**
- * Compares two regular expressions to see if they are equal.
+ * Compares two regular expressions or other types of conditions to see if they are equal.
  *
- * @param x - The first regular expression to compare.
- * @param y - The second regular expression to compare.
- * @returns True if the regular expressions are equal, false otherwise.
+ * @param x - The first condition to compare. It can be a string, a RegExp, a function that takes a string and returns a boolean, an array of RuleSetConditionAbsolute, or undefined.
+ * @param y - The second condition to compare. It is always a RegExp.
+ * @returns True if the conditions are equal, false otherwise.
  *
  * @remarks
- * This function compares two regular expressions to see if they are equal in terms of their source,
- * global, ignoreCase, and multiline properties. It is used to check if two regular expressions match
- * the same pattern.
+ * This function compares two conditions to see if they are equal in terms of their source,
+ * global, ignoreCase, and multiline properties. It is used to check if two conditions match
+ * the same pattern. If the first condition is not a RegExp, the function will always return false.
  */
 export const regexEqual = (
   x:
```

Comment:
The function 'regexEqual' is not implemented in this diff. It's mentioned that if the first condition is not a RegExp, the function will always return false. It's better to throw an error or a warning if the first condition is not a RegExp, to avoid silent failures.